src/System.Private.ServiceModel/tools/test/Bridge/Bridge/Program.cs

```diff
@@ -253,56 +284,90 @@ private static void ResetBridge(CommandLineArguments commandLineArgs)
             BridgeController.ReleaseAllResources(force: false);
         }
 
-        // Checks whether the Bridge is running and starts it if necessary
-        private static void RequireBridge(CommandLineArguments commandLineArgs)
+        // Checks whether the Bridge is running and starts it if necessary.
+        // Returns the exit code to return to the calling script
+        private static int RequireBridge(CommandLineArguments commandLineArgs)
         {
             string errorMessage = null;
+            Dictionary<string, string> properties;
 
-            if (PingBridge(commandLineArgs.BridgeConfiguration.BridgeHost,
-                                           commandLineArgs.BridgeConfiguration.BridgePort,
-                                           out errorMessage))
+            // In a test situation, the Bridge may have been launched and is still in the
+            // process of starting.  A successful ping means it is healthy, but we will retry
+            // if the ping fails and we detect the Bridge process is running.
+            DateTime startTime = DateTime.Now;
+            while (((DateTime.Now - startTime).TotalSeconds) < commandLineArgs.RequireBridgeTimeoutSeconds)
             {
-                Console.WriteLine("The Bridge is already running.");
-                Environment.Exit(0);
+                // Ping first in case the Bridge is running already or on another machine
+                if (PingBridge(commandLineArgs.BridgeConfiguration.BridgeHost,
+                                               commandLineArgs.BridgeConfiguration.BridgePort,
+                                               out errorMessage,
+                                               out properties))
+                {
+                    BridgeConfiguration bridgeConfiguation = new BridgeConfiguration(properties);
```

Comment:
Sorry could you clarify?  What don't we need and why don't we need it?  The initial ping allows someone to start the Bridge on another machine and let ToF naturally detect and use it.